runtime/near-vm-runner/src/vm_kind.rs

```diff
@@ -9,13 +9,13 @@ pub enum VMKind {
     /// Wasmtime VM.
     Wasmtime,
     /// Wasmer 1.x VM.
-    Wasmer1,
+    Wasmer2,
 }
```

Comment:
Let's preserve serialized format here by explicitelly changing discriminants:

```rust
#[derive(Clone, Copy, Debug, Hash, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
pub enum VMKind {
    Wasmer0 = 0,
    Wasmtime = 1,
//  Wasmer1 = 2,
    Wasmer2 = 3,
}
```

Let's also remove `Deserialize` and `BorshDeserialize` -- I don't belive they are actually used anywhere (transitive structs can have them removed as well), and that'll statically guarantee that removing vm kind won't blow up.

Reply:
Ouch, turns out bors ignores explicit discrimintants:

```rust
Ouch, so turns out that `borsh` ignores explicit discrimintants: 

```rust
#[derive(Clone, Copy, Debug, Hash, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
pub enum VMKind {
    /// Wasmer 0.17.x VM.
    Wasmer0 = 0,
    /// Wasmtime VM.
    Wasmtime = 1,
    /// Wasmer 1.x VM.
    Wasmer1 = 92,
}

#[test]
fn smoke() {
    let v = VMKind::Wasmer1;
    let data = v.try_to_vec().unwrap();
    assert_eq!(data, vec![2]); // 2, not 92
}
```

So this doesn't actually presereve serialization fromat. Let's remove the discrimintant then, and either leave an unused `wasmer1` or (if that's possible) just make a serialization breaking change here.